lib/site_settings/validations.rb

```diff
@@ -53,6 +53,10 @@ def validate_enable_s3_uploads(new_val)
     validate_error :s3_upload_bucket_is_required if new_val == "t" && SiteSetting.s3_upload_bucket.blank?
   end
 
+  def validate_enable_s3_inventory(new_val)
+    validate_error :s3_inventory_bucket_is_required if new_val == "t" && (SiteSetting.s3_inventory_bucket.blank? || SiteSetting.s3_upload_bucket.blank?)
+  end
+
```

Comment:
It's probably a good idea to add a validation to prevent bucket reuse.

https://github.com/discourse/discourse/blob/c928386c3ec6499a608363f855f8713bf4c4f2c9/lib/site_settings/validations.rb#L66-L72